### ISSUE TITLE
Tooltip: improve accessibility

### ISSUE THREAD

**Issue by Eehol** [primefaces/primefaces#5047]
## 1) Environment
- PrimeFaces version: 7.0
- Application server + version: any
- Affected browsers: any

## 2) Expected behavior

The tooltip should be accessible, 4 changes should make the tooltip compliant with level AA of WCAG2.1 standards.

2 Attributes should be added to the generated code ([source](https://www.w3.org/TR/wai-aria-practices-1.1/#tooltip)) :
- The div containing the text of the tooltip should have the attribute `role="tooltip"`
- The element described by the tooltip should have the attribute `aria-describedby="idOfTooltip" `

And 2 features are lacking, I don't know if it need to be activated by default or activated by using a new attribute on` p:tooltip `([source](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html)) :
- The content of the tooltip should be interactive ; it means that I should be able to move my mouse from the base text/link to the tooltip content without it being closed, for example to copy paste its content or to make a screen reader read it 
- The tooltip should be closable without moving out the mouse cursor or the focus, for example by pressing escape key

## 3) Actual behavior

The attributes role and describedby are not present, I can't hover the content of the tooltip without it being closed and I can't close the tooltip otherwise than by moving out the cursor or the focus.

## 4) Steps to reproduce

Just use `p:tooltip` with any attribute

## 5) Sample XHTML

```
<h:outputText value="Content to hover" id="texthover" />
<p:tooltip for="texthover" value="I'm a tooltip"  />
```

## 6) Sample bean

Not needed

**Comment by melloware** [comment]
@MarkusChur thoughts?

@Eehol I notice in the doc you posted this is not official yet?

> NOTE: This design pattern is work in progress; it does not yet have task force consensus. Progress and discussions are captured in issue 128.

**Comment by Eehol** [comment]
@melloware indeed it seems that only **if** [`role="tooltip"`](https://www.w3.org/TR/wai-aria/#tooltip) is used, **then** the `describedby `attribute should be present.  So you are right, it doesn't look mandatory at the moment, 

Nevertheless, the [WCAG rule 1.4.13](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html) references the adding of `role="tooltip"` in the sufficient techniques so it looks like it's already a good practice to me, but I am not an expert in accessibility practices so I could be wrong.

**Comment by MarkusChur** [comment]
# Summary - just my thoughts
* use provided javascript to achieve desired level of control
* add aria-describedby manually
* fix primefaces tooltip so that it does not disappear when moving the mouse onto the tooltip. 

# TL;DR
Consider the primefaces showcase: https://www.primefaces.org/showcase/ui/overlay/tooltip/tooltipOptions.xhtml

There is no way for blind people to consume the information of the tooltips. The reason is that a screen reader simply does not read out / announce these tooltips if they appear. By adding `aria-describedby` to the link/input, the tooltip text can be read out by focussing the element. However this can be a mess in combinations with labels. But it's what you make out of it, right? I don't think that primefaces must fix this. You can simply add this manually.

A keyboard only user is able to see tooltips if using `showEvent="focus"`. In all other cases it's inaccessible here.

In this case `showEvent="focus"` also fixes the issue that when moving the mouse outside the input, or onto the tooltip, you are still able to see the tooltip. The tooltips of the links fail here as the tooltip immediately disappears. The success criterion does not force interaction!

> For each area of additional content that appears on pointer hover:
The pointer can be moved over the new content without the additional content disappearing.
If this is false, then content fails the Success Criterion.

**Comment by melloware** [comment]
@MarkusChur So are you good with my PR just adding role="tooltip" to the DIVS of the tooltip?

**Comment by MarkusChur** [comment]
Maybe we can improve this by adding` aria-describedby` to the target. The question is if there are cases where this breaks things. It should work if we treat the attribute correctly. 
https://github.com/melloware/primefaces/blob/35e10807617686bef51866689ae97ae5f3150372/src/main/resources/META-INF/resources/primefaces/tooltip/tooltip.js#L155
If you add this here the target gets the correct `aria-describedby`. But this does not consider global tooltips.
`var describedBy = this.target.attr("aria-describedby");`
`if (describedBy != null && describedBy != "")`
`{`
`describedBy += " ";`
`describedBy += this.id;`
`}`
`else`
`{`
`describedBy = this.id;`
`}`
`this.target.attr("aria-describedby", describedBy);`

For global tooltips this can be done similiar in `bindGlobal` but we don't have a `id` for the `div` yet

**Comment by melloware** [comment]
I think we should only do it only for the singular tooltip.  The global tooltip to me would introduce too many issues trying to put describedby on every single component on a page which may have hundreds.

**Comment by MarkusChur** [comment]
totally makes sense.

**Comment by melloware** [comment]
OK PR has been updated i modified your code slightly.

**Comment by MarkusChur** [comment]
Looks good to me! Nice improvement of my snippet.
